packages/camera/camera_web/lib/src/camera.dart

```diff
@@ -171,4 +174,69 @@ class Camera {
       ..objectFit = 'cover'
       ..transform = 'scaleX(-1)';
   }
+
+  html.MediaRecorder? _mediaRecorder;
+  final StreamController<VideoRecordedEvent> _videoRecorderController =
+      StreamController();
+
+  /// Returns a Stream that emits when a video Recodring with a defined maxVideoDuration was created
+  Stream<VideoRecordedEvent> get onVideoRecordedEventStream =>
+      _videoRecorderController.stream;
+
+  /// Starts a new Video Recording using [html.MediaRecorder]
+  /// /// Throws a [html.DomException.INVALID_STATE] if there already is an active Recording
+  Future<void> startVideoRecording({Duration? maxVideoDuration}) async {
+    if (_mediaRecorder != null && _mediaRecorder!.state != 'inactive') {
+      throw html.DomException.INVALID_STATE;
+    }
+    _mediaRecorder ??= html.MediaRecorder(
+        videoElement.captureStream(), {'mimeType': 'video/webm'});
+
+    if (maxVideoDuration != null) {
+      _mediaRecorder!.addEventListener('dataavailable', (event) {
+        final blob = (event as html.BlobEvent).data;
+        final file = XFile(html.Url.createObjectUrl(blob));
+        _videoRecorderController
+            .add(VideoRecordedEvent(this.textureId, file, maxVideoDuration));
+        _mediaRecorder!.stop();
+      });
+      _mediaRecorder!.start(maxVideoDuration.inMilliseconds);
+    } else {
+      _mediaRecorder!.start();
+    }
+  }
+
+  /// Pauses the current video Recording
+  /// Throws a [html.DomException.INVALID_STATE] if there is no active Recording
+  Future<void> pauseVideoRecording() async {
+    if (_mediaRecorder == null || _mediaRecorder!.state == 'inactive') {
+      throw html.DomException.INVALID_STATE;
+    }
+    _mediaRecorder?.pause();
+  }
+
+  /// Resumes a video Recording
+  /// Throws a [html.DomException.INVALID_STATE] if there is no active Recording
+  Future<void> resumeVideoRecording() async {
+    if (_mediaRecorder == null || _mediaRecorder!.state == 'inactive') {
+      throw html.DomException.INVALID_STATE;
+    }
+    _mediaRecorder?.resume();
+  }
+
+  /// Stops the video Recording and will return the video as a webm video
+  /// Throws a [html.DomException.INVALID_STATE] if there is no active Recording
+  Future<XFile> stopVideoRecording() async {
+    if (_mediaRecorder == null || _mediaRecorder!.state == 'inactive') {
+      throw html.DomException.INVALID_STATE;
+    }
+    final availableData = Completer<XFile>();
+    _mediaRecorder!.addEventListener('dataavailable', (event) {
+      final blob = (event as html.BlobEvent).data;
+      availableData.complete(XFile(html.Url.createObjectUrl(blob)));
+    });
+    _mediaRecorder?.stop();
```

Comment:
There seems to be some code duplication here vs startVideoRecording (with a max length).

Could you unify these two codepaths so the implementation of the`dataavailable` event on `startVideoRecording` uses the same logic as "stop"?

I also wonder why the start with max length emits a `VideoRecordedEvent`, but `stopVideoRecording` doesn't?

Reply:
Unified it by registering the listener on `dataavailable` only in the startVideoRecording.

If the listener is triggered it will do the following things:
- emit a `VideoRecordedEvent`
- Complete the Completer used to obtain the `XFile` in `stopVideoRecording`
- remove the listener
- stop the MediaRecorder
   - This stop is necessary to only get data once if a `maxVideoDuration` is provided as `MediaRecorder` only takes in splices instead of a maxDuration